pkg/supportbundle/spec.go

```diff
@@ -106,106 +106,200 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
+	supportBundle := b.DeepCopy()
+
+	// determine an upload URL
+	var uploadURL string
+	var redactURL string
+	randomBundleID := strings.ToLower(rand.String(32))
+	if opts.DisableUpload {
+		//Just use the library internally
+		return supportBundle
+	} else if opts.Origin != "" {
+		uploadURL = fmt.Sprintf("%s/api/v1/troubleshoot/%s/%s", opts.Origin, app.GetID(), randomBundleID)
+		redactURL = fmt.Sprintf("%s/api/v1/troubleshoot/supportbundle/%s/redactions", opts.Origin, randomBundleID)
+	} else if opts.InCluster {
+		uploadURL = fmt.Sprintf("%s/api/v1/troubleshoot/%s/%s", fmt.Sprintf("http://kotsadm.%s.svc.cluster.local:3000", util.PodNamespace), app.GetID(), randomBundleID)
+		redactURL = fmt.Sprintf("%s/api/v1/troubleshoot/supportbundle/%s/redactions", fmt.Sprintf("http://kotsadm.%s.svc.cluster.local:3000", util.PodNamespace), randomBundleID)
+	} else {
+		uploadURL = fmt.Sprintf("%s/api/v1/troubleshoot/%s/%s", os.Getenv("API_ADVERTISE_ENDPOINT"), app.GetID(), randomBundleID)
+		redactURL = fmt.Sprintf("%s/api/v1/troubleshoot/supportbundle/%s/redactions", os.Getenv("API_ADVERTISE_ENDPOINT"), randomBundleID)
 	}
-	existingSecret.Data[SpecDataKey] = renderedSpec
-	existingSecret.ObjectMeta.Labels = labels
 
-	_, err = clientset.CoreV1().Secrets(util.PodNamespace).Update(context.TODO(), existingSecret, metav1.UpdateOptions{})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to update support bundle secret")
+	supportBundle.Spec.AfterCollection = []*troubleshootv1beta2.AfterCollection{
+		{
+			UploadResultsTo: &troubleshootv1beta2.ResultRequest{
+				URI:       uploadURL,
+				Method:    "PUT",
+				RedactURI: redactURL,
+			},
+		},
 	}
 
-	// Include discovered support bundle specs. Perform this action here so
-	// as not to add discovered specs to the default support bundle spec secret.
-	return addDiscoveredSpecs(supportBundle, app, clientset), nil
+	supportBundle = addDefaultDynamicTroubleshoot(supportBundle, app, imageName, pullSecret)
+	supportBundle = populateNamespaces(supportBundle, namespacesToCollect, namespacesToAnalyze)
+	supportBundle = deduplicatedCollectors(supportBundle)
+	supportBundle = deduplicatedAnalyzers(supportBundle)
+
+	return supportBundle
 }
 
-// injectDefaults injects the kotsadm default collectors/analyzers in the the support bundle specification.
-func injectDefaults(app apptypes.AppType, b *troubleshootv1beta2.SupportBundle, opts types.TroubleshootOptions, namespacesToCollect []string, namespacesToAnalyze []string) (*troubleshootv1beta2.SupportBundle, error) {
+// createClusterSpecificSupportBundle creates a support bundle spec with only cluster specific collectors, analyzers and upload result URI.
+func createClusterSpecificSupportBundle(app apptypes.AppType, b *troubleshootv1beta2.SupportBundle, opts types.TroubleshootOptions, namespacesToCollect []string, namespacesToAnalyze []string, imageName string, pullSecret *troubleshootv1beta2.ImagePullSecrets) *troubleshootv1beta2.SupportBundle {
```

Comment:
looks like this var isn't used
```suggestion
func createClusterSpecificSupportBundle(app apptypes.AppType, opts types.TroubleshootOptions, namespacesToCollect []string, namespacesToAnalyze []string, imageName string, pullSecret *troubleshootv1beta2.ImagePullSecrets) *troubleshootv1beta2.SupportBundle {
```

Reply:
removed